cli/asc.js

```diff
@@ -1165,6 +1171,64 @@ function checkDiagnostics(program, stderr) {
         EOL + EOL
       );
     }
+    if (reportDiagnostic) {
+      const diagnostic = __wrapDiagnosticMessage(diagnosticPtr);
+      __pin(diagnostic.message);
```

Comment:
Pinning here can be reduced to just `__pin(diagnostic)` while it is used, as the diagnostic object itself keeps all its members alive. A single `__unpin(diagnostic)` at the end will do then as well.

Reply:
oh, good to know :)